dev-tools/mage/godaemon.go

```diff
@@ -68,5 +75,12 @@ func BuildGoDaemon() error {
 // CrossBuildGoDaemon cross-build the go-daemon binary using the
 // golang-crossbuild environment.
 func CrossBuildGoDaemon() error {
-	return CrossBuild(ForPlatforms("linux"), WithTarget("buildGoDaemon"))
+	return CrossBuild(defaultCrossBuildGoDaemon...)
+}
+
+// CrossBuildGoDaemonWithSelectableImage cross-builds the go-daemon binary
+// using the selected image.
+func CrossBuildGoDaemonWithSelectableImage(f ImageSelectorFunc) error {
```

Comment:
Instead of adding a new function I think you could change `CrossBuildGoDaemon` to be variadic and pass options that are appended to the defaults. Like `func CrossBuildGoDaemon(opts ...CrossBuildOption) error`.